### PR TITLE
lesson 1 exercises

### PR DESCRIPTION
**Pull request by bapandaOK44**
README for my student working directory and lesson 1 exercises

### CODE REVIEW

File: students/barb_matthews/lesson2/gridPrinter.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#import math
+plus = '+ '
+minus = '- '
+pipe = '|'
+
+count = 4
+#grid = math.sqrt(count)
+grid = 2
+
+print (plus + minus*count + plus, end='')
```
Comment (charlesrobison) [inline students/barb_matthews/lesson2/gridPrinter.py:14]: No space after function name: print(...

File: students/barb_matthews/lesson2/gridPrinter.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+count = 4
+#grid = math.sqrt(count)
+grid = 2
+
+print (plus + minus*count + plus, end='') 
+print (minus*count + plus)
+
+while grid:
+
+	i=count
```
Comment (charlesrobison) [inline students/barb_matthews/lesson2/gridPrinter.py:19]: Missing spaces around assignment operator

File: students/barb_matthews/lesson2/gridPrinter.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	while i>0 :
+		cell = pipe + ' '*count*2 + ' '
+		print (cell*3)
+		#print()
+		i-=1
```
Comment (charlesrobison) [inline students/barb_matthews/lesson2/gridPrinter.py:24]: Need spaces here too

**Review by charlesrobison — APPROVED** [review]
Nice start Barb!  Recommend adding a linter to your text editor to catch PEP8 style issues.  Let me know if you have questions.

Ready to merge

**Comment by bapandaOK44** [comment]
What’s a linter? I like to use vi from iTerm to get practice, but also using Sublime Text. Is there a way to setup Sublime as a full IDE with debugger capabilities to step through code? All I’ve used was MS Visual Studio years ago and none of the new stuff. There’re so many fun things to explore :)

**Comment by natasha-aleksandrova** [comment]
@bapandaOK44 linter is a program that analyzes your code for various best practices violations..

Pycharm IDE is highly recommended as it comes built-in with all of the tools and best practices built-in.

Using command line editor while learning programming/new language is least desired as you don't get any built-in tools to help you write clean code - even though it is a great tool to know!

Yes, you can setup Sublime to be a full functional IDE for python.. but you need to spend some time to install all of the needed plugins that you would need, especially when you don't know all that need - that is why Pycharm is ideal. I use Pycharm for work and it is one of the best IDEs for python.

**Pull request merged**